File: JobMatchingSystem.API/Models/Company.cs
using JobMatchingSystem.API.Enums;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace JobMatchingSystem.API.Models
{
    public class Company
    {
        [Key]
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Description { get; set; }
        public string Logo { get; set; }
        public string Email { get; set; }
        public string? Website { get; set; }
        public string Address { get; set; }
        public string PhoneContact { get; set; }
        public string TaxCode { get; set; }
        public string LicenseFile { get; set; }

        public CompanyStatus Status { get; set; } = CompanyStatus.Pending;

        public string? RejectReason { get; set; }

        public int Score { get; set; } = 100;

        public int? VerifiedBy { get; set; }

        public DateTime? VerifiedAt { get; set; }

        public DateTime? CreatedAt { get; set; }

        public DateTime? UpdatedAt { get; set; }

        public bool IsActive { get; set; } = false;

        // Navigation properties
        [ForeignKey("VerifiedBy")]
        public virtual ApplicationUser? VerifiedByUser { get; set; }
        public virtual ICollection<ApplicationUser> ApplicationUsers { get; set; } = new List<ApplicationUser>();
        public virtual ICollection<Job> Jobs { get; set; } = new List<Job>();
    }
}

File: JobMatchingSystem.FE/src/models/company.ts
export interface Company {
  id: number;
  name: string;
  description: string;
  logo: string;
  email: string;
  website: string;
  address: string;
  phoneContact: string;
  status: number; // 0: Pending, 1: Approved, 3: Rejected
  taxCode: string;
  licenseFile: string;
  isActive: boolean; 
  score: number;
  rejectReason: string | null;
  createdAt: string;
  jobCount?: number; // Số tin tuyển dụng
  teamMembersCount?: number; // Số thành viên công ty
  recruitsCount?: number; // Số người đã tuyển
}

// Helper để convert status number thành string để hiển thị
export const getStatusString = (status: number): 'Pending' | 'Approved' | 'Rejected' => {
  switch (status) {
    case 0:
      return 'Pending';
    case 1:
      return 'Approved';  
    case 2:
      return 'Rejected';
    default:
      return 'Pending';
  }
};

// Helper để convert status string thành number để gửi API
export const getStatusNumber = (status: 'Pending' | 'Approved' | 'Rejected'): number => {
  switch (status) {
    case 'Pending':
      return 0;
    case 'Approved':
      return 1;
    case 'Rejected':
      return 2;
    default:
      return 0;
  }
};